.github/workflows/ci.yml

```diff
@@ -0,0 +1,105 @@
+name: Python CI
+
+on:
+  push:
+    branches:
+      - master
+    tags: ['*']
+  pull_request:
+
+concurrency:
+  group: test-${{ github.head_ref }}
+  cancel-in-progress: true
+
+env:
+  PYTHONUNBUFFERED: "1"
+  FORCE_COLOR: "1"
+
+jobs:
+  build:
+    name: Python ${{ matrix.python-version }} ${{ matrix.extra-env }}
+    if: >-
+        !contains(github.event.head_commit.message, '[skip ci]')
+        && !contains(github.event.head_commit.message, '[skip tests]')
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10", "3.11"]
+        include:
+          - python-version: "3.7"
+          - python-version: "3.7"
+            extra-env: "PIP_EXTRA_INSTALLATION='numpy symengine==0.5.0' USE_SYMENGINE=1"
```

Comment:
Is `extra-env` being used anywhere? I don't see environment variables being set anywhere, nor documentation for `extra-env` being part of github.

Reply:
I used it to install sympyengine which is supposed  to be picked up by SymPy for performance improvement (not much in the past). I need to check if it’s still effective nowadays.

Reply:
Please note that the `symengine` version is far behind the latest `0.10.0`, may need a seperate PR to bump it and check if it still works.

Reply:
OK, I put together what happened behind this line:

1. I want to see if a C++ backend of SymPy and improve performance
2. I check the documents to learn how to install `symengine` and set `USE_SYMENGINE=1` to activate the usage
3. According to what I read from https://github.com/sympy/sympy/blob/master/sympy/core/backend.py , I thought those impored functions will be replaced by `symengine` all over SymPy
4. I make the tests work (I'm under the impression that something went wrong in the process but fixed somehow) and found no performance improvements and thought it was because `symengine` isn't mature back then
5. Now I've just checked the docs and sources (by searching `USE_SYMENGINE` and `import backend` in `sympy`) and `backend` is only explicitly used in `sympy.physics.mechanics` and
  `sympy.liealgebras` modules which are NOT used by GAlgebra at all.

Feel free to remove references to `symengine` as this is a historical misunderstanding and the situation is still not improved, so CI run would be lighter.

Reply:
LOL.  I got `extra-env` from chatting with github copilot.  Thought I checked it, but not well enough.  This is the first time I've fallen for an AI hallucination!

Anyway, I've simplified the job matrix.